ophyd/ophydobj.py

```diff
@@ -52,28 +54,55 @@ def _finished(self, *args, **kwargs):
         #     print("this should be empty: {}".format(kwargs))
         self.done = True
 
-        if self._cb is not None:
-            self._cb()
-            self._cb = None
+        for cb in list(self._callbacks):
+            cb()
+            self._callbacks.remove(cb)
 
     @property
-    def finished_cb(self):
+    def callbacks(self):
         """
-        Callback to be run when the status is marked as finished
+        Callbacks to be run when the status is marked as finished
 
         The call back has no arguments
         """
-        return self._cb
+        return self._callbacks
 
-    @finished_cb.setter
     @_locked
-    def finished_cb(self, cb):
-        if self._cb is not None:
-            raise RuntimeError("Can not change the call back")
+    def add_callback(self, cb):
         if self.done:
             cb()
         else:
-            self._cb = cb
+            self._callbacks.add(cb)
+
+    def __and__(self, other):
+        composite = StatusBase()
+        composite.ancestors |= set([self, other])
+        composite.ancestors |= (self.ancestors | other.ancestors)
+
+        def inner():
+            with composite._lock:
+                composite._done_count += 1
```

Comment:
Why do `_done_count` and `ancestors` live in StatusBase? Isn't that something that should be in some `CompositeStatus` class? Currently, the result of a `(MoveStatus() | MoveStatus())` would give you a `StatusBase` as a result...